e2e/dractions/retry.go

```diff
@@ -143,3 +146,72 @@ func waitDRPCProgression(
 		}
 	}
 }
+
+func addDRPCAnnotation(ctx types.TestContext, drpcName string) error {
+	log := ctx.Logger()
+
+	return retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		drpc, err := getDRPC(ctx, ctx.ManagementNamespace(), drpcName)
+		if err != nil {
+			return err
+		}
+
+		if drpc.Annotations == nil {
+			drpc.Annotations = make(map[string]string)
+		}
+
+		drpc.Annotations[drpcDoNotDeletePVC] = "true"
+
+		if err := updateDRPC(ctx, drpc); err != nil {
+			return err
+		}
+
+		log.Debugf("Annotated drpc \"%s/%s\" with \"%s: %s\" in cluster %q",
+			ctx.ManagementNamespace(), drpcName, drpcDoNotDeletePVC,
+			drpc.Annotations[drpcDoNotDeletePVC], ctx.Env().Hub.Name)
+
+		return nil
+	})
+}
+
+func waitForDRPCAnnotationPropagation(ctx types.TestContext, drpcName string) error {
+	log := ctx.Logger()
+	start := time.Now()
+
+	log.Debugf("Waiting until drpc \"%s/%s\" annotation %q is propagated to primary vrg",
+		ctx.ManagementNamespace(), drpcName, drpcDoNotDeletePVC)
+
+	for {
+		drpc, err := getDRPC(ctx, ctx.ManagementNamespace(), drpcName)
+		if err != nil {
+			return err
+		}
+
+		preferredCluster, err := ctx.Env().GetCluster(drpc.Spec.PreferredCluster)
+		if err != nil {
+			return err
+		}
+
+		vrgName := drpcName
+		vrgNamespace := getVRGNamespace(ctx)
+
+		vrg, err := getVRG(ctx, preferredCluster, vrgNamespace, vrgName)
+		if err != nil {
+			return err
+		}
+
+		if vrg.Annotations[drpcDoNotDeletePVC] == "true" {
+			elapsed := time.Since(start)
+			log.Debugf("DRPC \"%s/%s\" annotation \"%s: true\" propagated to primary vrg \"%s/%s\" in cluster %q "+
+				" in %.3f seconds", ctx.ManagementNamespace(), drpcName, drpcDoNotDeletePVC,
+				vrgNamespace, vrgName, preferredCluster.Name, elapsed.Seconds())
+
+			return nil
+		}
+
+		if err := util.Sleep(ctx.Context(), util.RetryInterval); err != nil {
+			return fmt.Errorf("drpc \"%s/%s\" annotation \"%s: true\" propagation failed: %w",
```

Comment:
The message is not consistent with other wait functions, and it should mention the vrg, not the drpc. Maybe:

    "vrg "namespace/name" annotation "..." not propagated: %w"